### ISSUE TITLE
Helper_Functions_Test errors and GravityView_API_Test failure

### ISSUE THREAD

**Issue by soulseekah** [gravityview/GravityView#818]
```
Time: 34.26 seconds, Memory: 68.00MB

There were 2 errors:

1) GravityView_Helper_Functions_Test::test_gv_empty
Undefined variable: not_defined

/tests/unit-tests/helper-functions_Test.php:21

2) GravityView_Helper_Functions_Test::test_gravityview_number_format
A non well formed numeric value encountered

/src/wp-includes/functions.php:219
/tests/unit-tests/helper-functions_Test.php:155

--

There was 1 failure:

1) GravityView_API_Test::test_gravityview_get_current_views
The $current_views array didn't have a value set at $post->ID key
Failed asserting that false is true.

/tests/unit-tests/GravityView_API_Test.php:356

ERRORS!
Tests: 83, Assertions: 1216, Errors: 2, Failures: 1.
```

Against 1e56843 using PHPUnit 5.7, WordPress 4.7 and GravityForms HEAD at 976fd3ea (~Dec 23 master), on custom-built PHP 7.0.13 (cli) (built: Nov 18 2016 00:35:50) ( NTS ), stock 10.0.28-MariaDB, and Ubuntu 16.04.1 LTS (Linux 4.4.1).

Let me know if there's any other information I can provide.

**Comment by soulseekah** [comment]
Getting this on the latest master:

```
Time: 33.09 seconds, Memory: 68.00MB

There were 2 errors:

1) GravityView_Helper_Functions_Test::test_gv_empty
Undefined variable: not_defined

/tests/unit-tests/helper-functions_Test.php:21

2) GravityView_Helper_Functions_Test::test_gravityview_number_format
A non well formed numeric value encountered

/tests/unit-tests/helper-functions_Test.php:155

--

There was 1 failure:

1) GravityView_API_Test::test_gravityview_get_current_views
The $current_views array didn't have a value set at $post->ID key of 30
Failed asserting that false is true.

/tests/unit-tests/GravityView_API_Test.php:343

ERRORS!
Tests: 83, Assertions: 1215, Errors: 2, Failures: 1.
```

Slightly different failure now. CI appears to have reproduced the issue on all versions as well, strange...

**Comment by soulseekah** [comment]
OK. So I think I found the issue here. In `test_gravityview_get_current_views` the `GravityView_View_Data::getInstance` method gets called without the `passed_post` parameter. This instantiates an empty View_Data context without any views whatsoever. So we need to either:

- pass the `_get_new_view_id()` post like so: `GravityView_View_Data::getInstance( $post->ID )
- or set the context to `null` and add some content (perhaps with some view shortcodes) into the `$fe` instance (so it can parse it via its `parse_content()` method).

Ideally both cases would be tested, but going the path of least resistance and smallest changeset we'd have to pick the first fix. Will issue a pull request shortly.

Let's see what can be done about those errors now...

**Comment by soulseekah** [comment]
As for the second error:

```
2) GravityView_Helper_Functions_Test::test_gravityview_number_format
A non well formed numeric value encountered
```

This is triggered by the `number_format` built-in PHP function which is used by WordPress' `number_format_i18n` (and `number_format_i18n` is used by `gravityview_number_format`).

"1,000" is not a valid `float`.

`gravityview_number_format` needs to be reviewed and fixed as it does not operate as expected by the test, meanwhile those commas in the test have to be removed for it to not trigger the error and pass.

With this last commit I get `OK (83 tests, 1239 assertions)` :fire:

But the CI tests are still unstably failing with several unrelated errors.

**Comment by soulseekah** [comment]
Well, with all of the above + e7efeb8 we're finally getting some green builds on the latest version of WordPress:

https://travis-ci.org/gravityview/GravityView/builds/192468898

Issues with WordPress 4.0 remain. This might call for a nice git bisect job against core, exciting :hocho:

**Comment by soulseekah** [comment]
- 4.7 passes
- 4.6 passes
- 4.5 passes
- 4.4 passes
- 4.3 fails with:
```
There was 1 error:

1) GravityView_API_Test::test_gravityview_field_output
Undefined property: GravityView_API_Test::$atts

/tests/unit-tests/GravityView_API_Test.php:43
```
- 4.2 fails with:
```
There were 2 errors:

1) GravityView_API_Test::test_entry_link_html
Methods with the same name as their class will not be constructors in a future version of PHP; PasswordHash has a deprecated constructor

/tmp/wordpress-tests-lib/src/wp-includes/class-phpass.php:36
/tmp/wordpress-tests-lib/src/wp-includes/pluggable.php:1953
/tmp/wordpress-tests-lib/src/wp-includes/pluggable.php:1953
/tmp/wordpress-tests-lib/src/wp-includes/user.php:1827
/tests/gravityview-factory.php:102
/tests/gravityview-factory.php:92
/tests/gravityview-factory.php:140
/tests/unit-tests/GravityView_API_Test.php:207

2) GravityView_API_Test::test_gravityview_field_output
Undefined property: GravityView_API_Test::$atts

/tests/unit-tests/GravityView_API_Test.php:436
```
- 4.1 and 4.0 fail with:
```
There were 3 failures:

1) GravityView_Edit_Entry_Test::test_check_user_cap_edit_entry
gravityview_full_access
Failed asserting that false is true.

/tests/unit-tests/GravityView_Edit_Entry_Test.php:304
/tests/unit-tests/GravityView_Edit_Entry_Test.php:171

2) GravityView_Roles_Capabilities_Test::test_has_cap_gravityview_full_access
Failed asserting that false is true.

/tests/unit-tests/GravityView_Roles_Capabilities_Test.php:198

3) GravityView_Roles_Capabilities_Test::test_has_cap_single_post_cap
Failed asserting that false is true.

/tests/unit-tests/GravityView_Roles_Capabilities_Test.php:240
```

Not sure why we're not testing further back given that `GV_MIN_WP_VERSION` is 3.5 :rofl: but for now the above are priority for the CI to go green.

**Comment by soulseekah** [comment]
d16c562 fixes the `$atts` issue in 4.2 and 4.3.

Deprecation issues with 4.2 and lower in PHP7 in fixed in a29a0ce, where Travis will no longer test against it, just like HHVM.

Capability test failures fixed in 85d1fd9. A sort of bug in WordPress 4.2 and lower (trac ticket 28374).

**Comment by soulseekah** [comment]
[![Build Status](https://travis-ci.org/gravityview/GravityView.svg?branch=develop)](https://travis-ci.org/gravityview/GravityView)
